### PR TITLE
[#3712] Fix garbage assignments/operations/returns

### PR DESCRIPTION
**Pull request by jaking92**
Fixed a couple of places in code where potentially uninitialzed variables were having their values assigned or used in some way.

Passed Jenkins tests except for the failure described in #3689 on ub16-oracle.

### CODE REVIEW

File: server/core/src/rodsAgent.cpp [hunk 1]
```diff
@@ -250,14 +250,13 @@ int initRsCommFromServerSocket( rsComm_t *rsComm, int socket ) {
...
-        setLocalAddr( rsComm->sock, &rsComm->localAddr );
-        setRemoteAddr( rsComm->sock, &rsComm->remoteAddr );
+    if ( receiveSocketFromSocket( socket, &newSocket ) > 0 ) {
+        rsComm->sock = newSocket;
+        if ( 0 != rsComm->sock ) {
+            /* remove error messages from xmsLog */
+            setLocalAddr( rsComm->sock, &rsComm->localAddr );
+            setRemoteAddr( rsComm->sock, &rsComm->remoteAddr );
+        }
     }
```
Comment (trel) [inline server/core/src/rodsAgent.cpp]: need an else() here?

Comment (jaking92) [reply server/core/src/rodsAgent.cpp]: Upon closer inspection, looks like this function (initRsCommFromServerSocket) is not being referenced anywhere. We could fix it up or just remove it. Thoughts, @jasoncoposky?

Comment (trel) [reply server/core/src/rodsAgent.cpp]: not used anywhere... torch it.

**Comment by jaking92** [comment]
Jenkins tests passed for new commit.

**Pull request merged**